pkg/yttlibrary/data_values.go

```diff
@@ -95,3 +110,37 @@ func (v DataValues) checkNonDocs(val interface{}) error {
 
 	return nil
 }
+
+func processDVAnnotationArgs(annContent string) (map[string]string, error) {
```

Comment:
i think we should remove knowledge of library kwargs out of this place, which means DataValues.Extract() would continue to return single array of data values. lets move annotation checking (and splitting into two arrays) after templating, before overlaying over here: https://github.com/k14s/ytt/blob/dab244464dbdf009e59e81cabd9b6d9bf3b3747e/pkg/workspace/data_values_pre_processing.go#L53. then you can use template.HasAnnotation etc.